src/client/app/components/ExportComponent.tsx

```diff
@@ -2,213 +2,186 @@
 			const rateScaling = returned.needsRateScaling ? graphState.lineGraphRate.rate : 1;
 			// Loop over the displayed meters and export one-by-one.  Does nothing if no meters selected.
 			for (const meterId of graphState.selectedMeters) {
-				const meterArea = metersState[meterId].area;
+				const meterArea = meterDataById[meterId].area;
 				// export if area normalization is off or the meter can be normalized
-				if (!graphState.areaNormalization || (meterArea > 0 && metersState[meterId].areaUnit !== AreaUnitType.none)) {
+				if (!graphState.areaNormalization || (meterArea > 0 && meterDataById[meterId].areaUnit !== AreaUnitType.none)) {
 					// Line readings data for this meter.
-					const byMeterID = readingsState.line.byMeterID[meterId];
+					// Get the readings for the time range and unit graphed
+					const readingsData = lineMeterReadings[meterId];
 					// Make sure it exists in case state is not there yet.
-					if (byMeterID !== undefined) {
-						// Convert the meter area into the proper unit if normalizing by area or use 1 if not so won't change reading values.
-						const areaScaling = graphState.areaNormalization ?
-							meterArea * getAreaUnitConversion(metersState[meterId].areaUnit, graphState.selectedAreaUnit) : 1;
-						// Divide areaScaling into the rate so have complete scaling factor for readings.
-						const scaling = rateScaling / areaScaling;
-						// Get the readings for the time range and unit graphed
-						const byTimeInterval = byMeterID[timeInterval.toString()];
-						if (byTimeInterval !== undefined) {
-							const readingsData = byTimeInterval[unitId];
-							// Make sure they are there and not being fetched.
-							if (readingsData !== undefined && !readingsData.isFetching) {
-								if (readingsData.readings === undefined) {
-									throw new Error(`Unacceptable condition: readingsData.readings is undefined for meter ${meterId}.`);
-								}
-								// Get the readings from the state.
-								const readings = _.values(readingsData.readings);
-								// Sort by start timestamp.
-								const sortedReadings = _.sortBy(readings, item => item.startTimestamp, 'asc');
-								// Identifier for current meter.
-								const meterIdentifier = metersState[meterId].identifier;
-								graphExport(sortedReadings, meterIdentifier, unitLabel, unitIdentifier, chartName, scaling, MeterOrGroup.meter, errorBarState);
-							}
-						}
+					// Convert the meter area into the proper unit if normalizing by area or use 1 if not so won't change reading values.
+					const areaScaling = graphState.areaNormalization ?
+						meterArea * getAreaUnitConversion(meterDataById[meterId].areaUnit, graphState.selectedAreaUnit) : 1;
+					// Divide areaScaling into the rate so have complete scaling factor for readings.
+					const scaling = rateScaling / areaScaling;
+					// Make sure they are there and not being fetched.
+					if (readingsData) {
```

Comment:
Should this have as lineMeterIsFetching?

Reply:
Line 66 should cover this, but the other cases should also be carefully evaluated. I suppose a check that there is actually meaningful data like readings.length ... etc could be checked here as well. 

I didn't significantly alter the logic in this or most components,  except for how the data is retrieved. 
Upon careful inspection of this code, it appears that many of these checks can be deferred to some already existing selectors, and possibly re-written for easier maintenance which seams like a near-future endeavor.
For example, this line

A refactor should be in order for this and probably many other files.